similarity/calc_sim.py

```diff
@@ -0,0 +1,78 @@
+import nltk
+from nltk.corpus import stopwords
+import numpy as np
+import pandas as pd
+from sklearn.cluster import KMeans
+import string
+
+
+# Reads csv as input with Pandas DataFrame
+def read_data(filename):
+  data = pd.read_csv(filename, skipinitialspace=True)
+  return data
+
+# map each word to an index
+def extract_words(data):
+  bag_of_words = {}
+  nltk.download('stopwords')
+  stop_words = set(stopwords.words('english'))
+  index = 0
+  for about in data['Mission']:
+    # remove all punctuations
+    for char in about:
+      if char in string.punctuation:
+        about = about.replace(char, ' ')
+    # convert to all lowercase
+    about = about.lower()
+    # map each word to its index
+    for word in about.split():
+      if word not in bag_of_words.keys() and word not in stop_words:
```

Comment:
Here and throughout:

For a `dict`,

```python
word not in bag_of_words
```

is equivalent to

```python
word not in bag_of_words.keys()
```

The succinct version is more Pythonic.